controllers/metal3.io/hostfirmwaresettings_controller.go

```diff
@@ -205,13 +208,16 @@ func (r *HostFirmwareSettingsReconciler) updateStatus(info *rInfo, settings meta
 			continue
 		}
 
-		info.hfs.Status.Settings[k] = v
+		newStatus.Settings[k] = v
 	}
 
+	dirty = !reflect.DeepEqual(info.hfs.Status.FirmwareSchema, newStatus.FirmwareSchema) ||
+		!reflect.DeepEqual(info.hfs.Status.Settings, newStatus.Settings)
+
 	// Check if any Spec settings are different than Status
 	specMismatch := false
 	for k, v := range info.hfs.Spec.Settings {
-		if statusVal, ok := info.hfs.Status.Settings[k]; ok {
+		if statusVal, ok := newStatus.Settings[k]; ok {
```

Comment:
Q: IIUC, this potentially may reconcile two distinct facts, either that the user updated meanwhile the HFS specs, either that someone else changed directly the BIOS settings (maybe a manual operation?), can you please confirm? In any case, I guess it could be worth logging the event for future troubleshooting